### ISSUE TITLE
Support header sso (Forward Auth)

### ISSUE THREAD

**Issue by RandomExplosion** [nitnelave/lldap#352]
Since one common use case for lldap is as a ldap server for a separate authentication server such as authelia it would be good if the web ui was compatible with login from those providers e.g. [through trusted header sso](https://www.authelia.com/integration/trusted-header-sso/introduction/)

Currently if one wants to put lldap behind 2fa you can still use forward auth with a reverse proxy to block access to the dashboard if you haven't signed in with 2fa but you still need to sign-in again with your password after that.

Basically you can either have username+password or 2x username and password + 2fa
This works fine but does feel a bit clunky.

As I understand it it shouldn't be *too* difficult to implement either since it takes the place of the current login screen, either way I haven't read through all of lldap's source to know for sure.

**Comment by dvv** [comment]
@nitnelave 
I wonder if we could switch native auth off completely with some config option.
Then authenticating to lldap via authelia (using lldap per se) would be similar to securing vanilla whoami example.

**Comment by nitnelave** [comment]
@dvv How would you get the current user? The proposed way (that I approve) is through trusted headers, but do you have another proposal?

**Comment by dvv** [comment]
Exactly via trusted headers. I think if we remove native login and honor `Remote-User:` as the current legal user.

**Comment by nitnelave** [comment]
The backend changes are going to be straightforward (just parse the header if it's there), it's the front-end changes that are going to be a bit more challenging, I think: When the frontend loads, it doesn't know which user is logged in. It would have to do a request to the server to check "am I already logged in?" since it can't rely on a cookie.

**Comment by dvv** [comment]
Well, I believe non-http-only cookies should go away.
Probably there should be some "profile" API (RPC?) method that returns user info.
I didn't dive in client-side of lldap.

**Comment by nitnelave** [comment]
The only non-http-only cookies right now are storing who's logged in and whether they're admin, but they don't store any credentials (it's to help the frontend). There's a small chance they get out of sync with the actual logged in state (if the session was somehow invalidated from the backend), but so far it's worked out pretty well.

Another way to do it would be indeed to have a "whoami" endpoint that returns the currently logged in user, or none.

**Comment by fiinnnn** [comment]
Is anyone working on this currently? I'm happy to take a shot at implementing it if I have some time.

**Comment by nitnelave** [comment]
No one's working on it at the moment, it's all yours!

**Comment by tecosaur** [comment]
@fiinnnn did you take this for a spin? It would be quite cool to have this functionality :slightly_smiling_face:

**Comment by fiinnnn** [comment]
@tecosaur unfortunately I haven't been able to find the time to look at this so far, if anyone else would like to try definitely go for it

**Comment by tecosaur** [comment]
Ah well, that's a bit of a pity but understandable. Thanks for the update!

**Comment by nitnelave** [comment]
Note that propagation of settings to the frontend (#514 ) is done, so this should be easier.

**Comment by Kumpelinus** [comment]
FYI: I started working on this Issue

**Comment by CumpsD** [comment]
Linking the follow up PR in here: https://github.com/lldap/lldap/pull/1429

Would love to have this too, I have authelia in front of everything with traeffik and try to get sso everywhere :)
